api/gateway/share_test.go

```diff
@@ -0,0 +1,98 @@
+package gateway
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_dataFromShares(t *testing.T) {
+	type testCase struct {
+		name    string
+		input   [][]byte
+		want    [][]byte
+		wantErr bool
+	}
+
+	smallTxInput := padShare([]uint8{
```

Comment:
[for reviewers] this unit test is tightly coupled to the share format defined in celestia-app so I can remove these unit tests if maintainers think celestia-node should treat the shares returned from celestia-app as opaque